### ISSUE TITLE
Mempack data corruption in libgit 1.6.3

### ISSUE THREAD

**Issue by edrevo** [libgit2/libgit2#6543]
### Reproduction steps
The following Rust code passes in git2-rs 0.16.1 (which uses libgit2 1.5.1) but fails on 0.17.0 (which uses libgit2 1.6.3)

```rust
    #[test]
    fn test_mempack_issue() -> Result<()> {
        let odb = git2::Odb::new()?;
        odb.add_new_mempack_backend(1)?;
        let repo = git2::Repository::from_odb(odb)?;

        //let tmpdir = tempdir::TempDir::new("")?;
        //let repo = git2::Repository::init_bare(tmpdir.path())?;
        let odb = repo.odb()?;

        let file_oid = odb.write(git2::ObjectType::Blob, &[0, 0, 0, 0])?;
        let mut tb = repo.treebuilder(None)?;
        tb.insert("file1", file_oid, i32::from(git2::FileMode::Blob))?;
        tb.insert("file2", file_oid, i32::from(git2::FileMode::Blob))?;
        let tree_oid = tb.write()?;

        // now create a commit
        let root_tree = repo.find_tree(tree_oid)?;
        let signature = git2::Signature::now("anakin skywalker", "darth@vader.dev")?;
        let commit_oid = repo.commit(None, &signature, &signature, "test commit", &root_tree, &[])?;
        repo.find_commit(commit_oid)?;
        Ok(())
    }
```

```
Error: failed to parse bad commit object; class=Object (11)
```

If you comment out the commented code, the test passes, which seems to indicate that the problem lies with the mempack implementation.

### Expected behavior
Code should pass both in 1.6.3 and 1.5.1

### Operating system(s) tested
Mac OS X

**Comment by ethomson** [comment]
I believe this may have been fixed by #6553, can you confirm? 🙏

**Comment by ethomson** [comment]
Closing - let me know if #6553 didn't fix your problems. 🙏
